seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java

```diff
@@ -87,6 +89,8 @@ public class PhysicalPlan {
 
     private volatile boolean isRunning = false;
 
+    private final JobStateEventListner jobStateEventListner;
```

Comment:
we should reuse https://github.com/apache/seatunnel/blob/085023ad0db1a8729ce724a14855c0077f051dfe/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java#L187